pkg/agent/metrics/metrics.go

```diff
@@ -136,6 +174,9 @@ func (a *AgentMetrics) Reset() {
 	a.endpointConnections.Reset()
 	a.streamPackets.Reset()
 	a.streamErrors.Reset()
+	a.serverCount.Reset()
```

Comment:
Accumulating values clearly need to be reset. Less clear to me that polled values (like serverCount) make sense to be reset. Reading the value after resetting and getting the reset value doesn't mean that new servers haven't accumulated, it just means you haven't polled since the reset.

Reply:
Gotcha, I just assumed that all the metrics should be reset when this method is called.

Reply:
Done.